internal/k6runner/k6runner_test.go

```diff
@@ -404,13 +417,33 @@ func TestK6LogsToLogger(t *testing.T) {
 	require.NoError(t, err)
 }
 
-type testLogger struct{}
+type testLogger struct {
+	buf *bytes.Buffer
+}
 
 var _ logger.Logger = &testLogger{}
 
 func (l *testLogger) Log(keyvals ...any) error {
 	if len(keyvals) == 0 {
 		return errors.New("empty log message")
 	}
-	return nil
+
+	if len(keyvals)%2 != 0 {
+		return errors.New("not the same number of keys and vals")
+	}
+
+	if l.buf == nil {
+		return nil
+	}
```

Comment:
nit: Since we are effectively supporting two different behaviors for the `testLogger` depending on whether a buffer is set for it or not, should we add a comment for that as the `testLogger` description? E.g.: If no buffer is set it only verifies inputs, if a buffer is set it writes to the buffer.

Reply:
Thinking about it I might as well make two different implementation for this 🤔

Reply:
Done in the last fixup!